src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -28,7 +28,7 @@ class Dockerfile extends DefaultTask {
     File destFile = project.file("$project.buildDir/docker/Dockerfile")
 
     Dockerfile() {
-
+        inputs.property("instructionsCache", { instructions.collect { it.build() }.join('\n') })
```

Comment:
time passes by ... so my comment https://github.com/bmuschko/gradle-docker-plugin/issues/357#issuecomment-321316089 is now outdated, but perhaps still valid?!